integration-tests/src/tests/test_download_file.rs

```diff
@@ -5,33 +5,39 @@ use std::fs::File;
 use std::io::Read;
 use std::net::SocketAddr;
 
-#[test]
-fn test_file_download() {
+#[tokio::test]
+async fn test_file_download() {
     let port = portpicker::pick_unused_port().expect("No ports free");
 
     async fn handle_request(_req: Request<Body>) -> Result<Response<Body>, Infallible> {
         let data: [u8; 1024] = [42; 1024];
         Ok(Response::new(Body::from(data.to_vec())))
     }
 
-    std::thread::spawn(move || {
-        tokio::runtime::Runtime::new().unwrap().block_on(async move {
-            let addr = SocketAddr::from(([127, 0, 0, 1], port));
-            let make_svc =
-                make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(handle_request)) });
-            let server = Server::bind(&addr).serve(make_svc);
-            if let Err(e) = server.await {
-                eprintln!("server error: {}", e);
-            }
-        });
+    let (tx, rx) = tokio::sync::oneshot::channel();
```

Comment:
Doesn't matter in this case, but in general the rule-of-thumb primitive to use for synchronization in tests is barrier: https://docs.rs/tokio/latest/tokio/sync/struct.Barrier.html. It more directly expresses "wait util the coordinating threads have finished setup work", and generalizes for more than one thread.

Reply:
ah cool, that does seem more clear. might as well update it